src/auth/src/credentials/idtoken.rs

```diff
@@ -27,6 +66,207 @@ use std::sync::Arc;
 use std::time::{Duration, SystemTime, UNIX_EPOCH};
 use tokio::time::Instant;
 
+pub mod mds {
+    //! Credentials for authenticating with [ID tokens] using [Metadata Service].
+    //!
+    //! Google Cloud environments such as [Google Compute Engine (GCE)][gce-link],
+    //! [Google Kubernetes Engine (GKE)][gke-link], or [Cloud Run] provide a metadata service.
+    //! This is a local service to the VM (or pod) which (as the name implies) provides
+    //! metadata information about the VM. The service also provides access
```

Comment:
Since you said "VM (or pod)" above then this should be:

```suggestion
    //! metadata information about the environment. The service also provides access
```